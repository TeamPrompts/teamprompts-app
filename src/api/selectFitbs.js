import { ACTIVE, DESC, FITBS, ID, STATUS } from './constants';
import fitbs from '../mocks/fitbs';
import makeFitb from './makeFitb';
import select from './select';

const { REACT_APP_AIRPLAIN_MODE } = process.env;

function selectFitbs(callback) {
  if (REACT_APP_AIRPLAIN_MODE === 'on') {
    callback(null, fitbs);
  } else {
    select({
      callback,
      map: makeFitb,
      options: {
        filterByFormula: `${STATUS}="${ACTIVE}"`,
        sort: [{ field: ID, direction: DESC }]
      },
      tableName: FITBS
    });
  }
}

export default selectFitbs;
