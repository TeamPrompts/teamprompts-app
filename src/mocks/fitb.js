import { SEPARATOR } from '../constants';

export const SOURCE = `Do you know the way to ${SEPARATOR}?`;

const fitb = {
  examples: ['San Jose'],
  id: 'receM4Gus7HH43oLM',
  prompts: ['[place]'],
  source: SOURCE,
  tags: ['recTqD26ElVvXKVad']
};

export default fitb;
