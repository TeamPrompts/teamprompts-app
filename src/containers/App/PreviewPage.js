import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { ID } from '../../api/constants';
import selectFitbs from '../../api/selectFitbs';
import FillInTheBlanks from '../../FillInTheBlanks/FillInTheBlanks';
import Heading from './Heading';

function PreviewPage() {
  const { id } = useParams();

  const [error, setError] = useState();
  const [fitbs, setFitbs] = useState();
  const [waiting, setWaiting] = useState(true);

  useEffect(() => {
    selectFitbs({ filterByFormula: `${ID}="${id}"` }, (error, fitbs) => {
      if (error) {
        setError(error);
      } else {
        setFitbs(fitbs);
      }
    });
  }, [id]);

  useEffect(() => {
    if (error) {
      setWaiting(false);
    }
    if (fitbs) {
      setWaiting(false);
    }
  }, [error, fitbs]);

  return (
    <>
      <Heading title="TeamPrompts" />
      <ul>
        {!error &&
          !waiting &&
          fitbs.map(fitb => (
            <li key={fitb.id}>
              <FillInTheBlanks fitb={fitb} />
            </li>
          ))}
        {!error && !waiting && fitbs.length === 0 && (
          <li>
            <pre>404</pre>
          </li>
        )}
      </ul>
      {error && <pre>{JSON.stringify(error, 0, 2)}</pre>}
    </>
  );
}

export default PreviewPage;
