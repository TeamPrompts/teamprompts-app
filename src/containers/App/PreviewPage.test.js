import { render } from '@testing-library/react';
import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import selectFitbs from '../../api/selectFitbs';
import fitb from '../../mocks/fitb';
import PreviewPage from './PreviewPage';

jest.mock('../../api/selectFitbs');

const mockId = fitb.id;
// const mockSlug = tag.slug;

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: mockId
  })
}));

describe('PreviewPage', () => {
  beforeEach(() => {
    selectFitbs.mockImplementation((options, callback) =>
      callback(null, [fitb])
    );
  });

  it('w/ fitb', () => {
    const { container } = render(
      <Router>
        <PreviewPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it('w/ fitb not found', () => {
    selectFitbs.mockImplementation((options, callback) => callback(null, []));

    const { container } = render(
      <Router>
        <PreviewPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it('w/ error', () => {
    selectFitbs.mockImplementation((options, callback) =>
      callback(new Error('Oh Noes!'))
    );

    const { container } = render(
      <Router>
        <PreviewPage />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
