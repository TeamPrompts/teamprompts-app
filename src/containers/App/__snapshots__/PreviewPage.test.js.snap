// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewPage w/ error 1`] = `
<div>
  <h1
    class="font-bold mb-8 mt-2 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>
  <ul />
  <pre>
    {}
  </pre>
</div>
`;

exports[`PreviewPage w/ fitb 1`] = `
<div>
  <h1
    class="font-bold mb-8 mt-2 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>
  <ul>
    <li>
      <div
        class="border-b border-gray-500 leading-loose mb-2 px-4 py-8 shadow text-lg"
        data-testid="container"
      >
        The weakest link in our mental model is 
        <span
          class="inline-flex flex-col leading-none"
          data-testid="input-container"
        >
          <div
            class="self-center"
            style="display: inline-block;"
          >
            <input
              class=""
              data-testid="input"
              placeholder="________________"
              style="box-sizing: content-box; width: 2px;"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            >
              ________________
            </div>
          </div>
          <label
            class="self-center text-sm"
          >
            some assumption
          </label>
        </span>
        . If weâ€™re off -- even by a little -- it could have a big impact on this initiative because 
        <span
          class="inline-flex flex-col leading-none"
          data-testid="input-container"
        >
          <div
            class="self-center"
            style="display: inline-block;"
          >
            <input
              class=""
              data-testid="input"
              placeholder="________________"
              style="box-sizing: content-box; width: 2px;"
              value=""
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            />
            <div
              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
            >
              ________________
            </div>
          </div>
          <label
            class="self-center text-sm"
          >
            reason
          </label>
        </span>
        .
        <span
          class="inline-block ml-3 relative"
        >
          <input
            class="hidden"
            data-testid="reckkQSV8NuHKOAHr"
            id="reckkQSV8NuHKOAHr"
            type="checkbox"
          />
          <label
            class="cursor-pointer"
            for="reckkQSV8NuHKOAHr"
          >
            <div
              class="w-6 h-3 bg-gray-400 rounded-full shadow-inner"
            />
            <div
              class="toggle__dot absolute w-4 h-4 bg-gray-300 rounded-full shadow inset-y-0 left-0"
            />
          </label>
        </span>
        <button
          class="bg-blue-500 focus:outline-none font-bold font-mono hover:bg-blue-700 ml-3 px-4 rounded-full text-xs text-white"
          data-testid="copy-button"
          type="button"
        >
          Copy to clipboard
        </button>
      </div>
    </li>
  </ul>
</div>
`;

exports[`PreviewPage w/ fitb not found 1`] = `
<div>
  <h1
    class="font-bold mb-8 mt-2 text-4xl text-gray-900"
  >
    TeamPrompts
  </h1>
  <ul>
    <li>
      <pre>
        404
      </pre>
    </li>
  </ul>
</div>
`;
